### PR TITLE
Add environment information utility

### PR DESCRIPTION
**Pull request by farhan-pasha**
# Add environment information utility

## Description

This commit introduces a new utility package for printing all the set environment variables and specific feature. It provides functionality to retrieve and print environment variables, collect additional system information, and ensure proper formatting of the output. This utility can be used by any image to capture and print its env information. Test coverage is 100%.

Resolves: #125 

## Changes

- Introduces new utility package which can be used by multiple images to print the env information.
- Updates the way version is being consumed in the docker file labels
- Adds unit tests

## Test Policy Compliance

- [X] I have added or updated unit tests for my changes.
- [ ] I have included integration tests where applicable.
- [X] All new and existing tests pass successfully.

## Code Quality

- [X] I have followed the coding standards for this project.
- [X] I have performed a self-review of my code.
- [X] My code is well-commented, particularly in areas that may be difficult 
      to understand.

## Documentation

- [X] I have made corresponding changes to the documentation (if applicable).
- [X] I have updated any relevant READMEs or wiki pages.

## Checklist

Before you submit this PR, please make sure:
- [X] You have read [the contributing guidelines][contributing] and 
      *especially* the [test policy][test-policy].
- [X] You have thoroughly tested your changes.
- [X] You have followed all the contributing guidelines for this project.
- [X] You understand and agree that your contributions will be publicly available 
  under the project’s license.

[contributing]: https://vsecm.com/docs/contributing/
[test-policy]: https://vsecm.com/docs/contributing/#add-tests-for-new-features

*By submitting this pull request, you confirm that my contribution is made under 
the terms of the project’s license and that you have the authority to grant 
these rights.*

---

Thank you for your contribution to [**VMware Secrets Manager**](https://vsecm.com)
🐢⚡️!

### CODE REVIEW

**Comment by farhan-pasha** [comment]
Example Output(for safe): shows all the set env vars and shows the values for selected features

Screenshot 2024-01-30 at 00 36 51

**Review by v0lkan — APPROVED** [review]
Good idea and great starting point.

I didn’t see anything catastrophic in the code.

A 🌮 from me :)

File: core/util/envinfo.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	"os"
+	"runtime"
+	"sort"
+	"strings"
+
+	"github.com/vmware-tanzu/secrets-manager/core/log"
+)
+
+// PrintEnvironmentInfo prints information about specific environment variables, enabled features, and binary version.
+func PrintEnvironmentInfo(id *string, envVarsToPrint []string) {
```
Comment (v0lkan) [inline core/util/envinfo.go:24]: I have mixed feelings about keeping this in Util versus moving it into the logging package. — Let’s keep it this way; but don’t be surprised if I refactor it elsewhere :) .

File: dockerfiles/example/init-container.Dockerfile [hunk 2]
```diff
@@ -22,8 +22,10 @@ RUN CGO_ENABLED=0 GOOS=linux go build -mod vendor -a -o example \
 # generate clean, final image for end users
 FROM gcr.io/distroless/static-debian11
 
+ENV APP_VERSION="0.21.2"
```
Comment (v0lkan) [inline dockerfiles/example/init-container.Dockerfile:25]: +1 for defining app version as an env var.

**Pull request merged**